src/migrateCalculatedMeasuresToActivePivot/migrateCalculatedMeasuresInWidgets.test.ts

```diff
@@ -0,0 +1,112 @@
+import { migrateCalculatedMeasuresInWidgets } from "./migrateCalculatedMeasuresInWidgets";
+import { uiWidgetsFolder } from "../__test_resources__/aui5.0LegacyTestResources/uiWidgetsFolder";
+import { dataModelsForTests } from "@activeviam/data-model";
+
+const dataModel = dataModelsForTests.sandbox;
+
+// "pvSum ^ 2" is from cube "EquityDerivativesCubeDist", all others are from "EquityDerivativesCube".
+const namesOfCalculatedMeasurestoMigrate = [
+  "Distinct count city",
+  "Test calculated measure",
+  "EXP pnl.Forex",
+  "Log pv.SUM",
+  "pvSum ^ 2",
+];
+
+describe("migrateCalculatedMeasuresInWidgets", () => {
+  const { cubeNames, migratedWidgetsRecord } =
+    migrateCalculatedMeasuresInWidgets(
+      uiWidgetsFolder,
+      dataModel,
+      namesOfCalculatedMeasurestoMigrate,
+    );
+
+  it("returns a `cubeNames` object containing the names of all calculated measures used in a `ui/widgets` folder with their corresponding cube name", () => {
+    expect(cubeNames).toStrictEqual({
+      "Distinct count city": "EquityDerivativesCube",
+      "EXP pnl.Forex": "EquityDerivativesCube",
+      "Log pv.SUM": "EquityDerivativesCube",
+      "Test calculated measure": "EquityDerivativesCube",
+      "pvSum ^ 2": "EquityDerivativesCubeDist",
+    });
+  });
+
+  it("returns the widget record unchanged if a widget at the root contains no calculated measures", () => {
+    // "7a5" is a widget containing no calculated measures at the root. This is definitely defined, otherwise the snapshot would be empty.
+    expect(
+      migratedWidgetsRecord.children?.content.children!["7a5"],
+    ).toStrictEqual(uiWidgetsFolder.children?.content.children!["7a5"]);
+  });
+
+  it("does not remove a calculated measure definition from the MDX if it is not on the list of `namesOfCalculatedMeasuresToMigrate`", () => {
+    // "ee7" is a widget containing 1 calculated measure which is not on the list of `namesOfCalculatedMeasuresToMigrate`. This is definitely defined, otherwise the snapshot would be empty.
+    expect(
+      migratedWidgetsRecord.children?.content.children!["ee7"],
+    ).toStrictEqual(uiWidgetsFolder.children?.content.children!["ee7"]);
+  });
+
+  it("removes the calculated measure definition from a widget containing 1 calculated measure at the root", () => {
```

Comment:
What does "at the root" mean here ? Same for other occurrences.
I would suggest to remove it, but if I'm missing something let me know.
```suggestion
  it("removes the calculated measure definition from a widget containing 1 calculated measure", () => {
```

EDIT: after reading below I think you mean "at the root" vs "in a folder", right ?
If so, I still think it's okay remove "at the root" and to just specify "widget in a folder" where needed.
Because if we keep the current wording I think most people would think 
 that "at the root" refers to the calculated measure, and wonder what that means.